### ISSUE TITLE
Repeat does not work properly

### ISSUE THREAD

**Issue by slackmoehrle** [cocos2d/cocos2d-x#15345]
consider the following: 

``` cpp
auto root = cocos2d::Node::create();
    addChild(root);

    root->runAction(CallFunc::create([&]() {
        CCLOG("repeat A");
    }));

    root->runAction(Repeat::create(CallFunc::create([&]() {
        CCLOG("repeat B");
    }), 1));

    root->runAction(Repeat::create(Sequence::createWithTwoActions(
        CallFunc::create([&]() { CCLOG("repeat C");}),
        DelayTime::create(0.25f)),
        1)
    );

    root->runAction(Repeat::create(Sequence::createWithTwoActions(
        CallFunc::create([&]() { CCLOG("repeat D");}),
        DelayTime::create(0.1f)),
        3)
    );
```

and the output:

```
repeat A
repeat C
repeat D
repeat D
repeat D
repeat D
```

No **repeat B** and breakpoint doesn't halt execution either. And **repeat D** repeats 4 times instead of 3.

**Comment by minggo** [comment]
I will take a look.

**Comment by minggo** [comment]
fix in #15348 
@slackmoehrle Could you please verify it?

**Comment by piotrros** [comment]
```
repeat A
repeat B
repeat C
repeat D
repeat D
repeat D
```

Nice, it's working :)
